include/pm/policy.h

```diff
@@ -33,6 +66,102 @@ const struct pm_state_info *pm_policy_next_state(uint8_t cpu, int32_t ticks);
 
 /** @endcond */
 
+#if defined(CONFIG_PM) || defined(__DOXYGEN__)
+/**
+ * @brief Allow a power state.
+ *
+ * A power state will be disallowed on the first call of
+ * pm_policy_state_disallow(). Subsequent calls will just increase a reference
+ * count, thus meaning this API can be safely used concurrently. A state will
+ * be allowed again after pm_policy_state_allow() is called as many times as
+ * pm_policy_state_disallow().
```

Comment:
considering one of the above examples - what will happen when user disallows all?

Reply:
clarified docs